packages/fxa-settings/src/models/Account.ts

```diff
@@ -771,13 +772,7 @@ export class Account implements AccountData {
       sessionToken(accountReset.sessionToken);
       return accountReset;
     } catch (err) {
-      const errno = (err as ApolloError).graphQLErrors[0].extensions?.errno as
-        | number
-        | undefined;
-      if (errno && AuthUiErrorNos[errno]) {
-        throw AuthUiErrorNos[errno];
-      }
-      throw AuthUiErrors.UNEXPECTED_ERROR;
+      throw getHandledError(err);
```

Comment:
This update is what fixed the wrong error messaging. Since I imagine we're going to want to use this elsewhere (I think we can also remove a function in SignUp utils to use this instead, but didn't do that here), I went ahead and moved it out from `Signin` and in `lib`.